neuro_san/internals/run_context/langchain/journaling/journaling_callback_handler.py

```diff
@@ -151,6 +151,10 @@ async def on_tool_start(
             combined_args["origin"] = self.origin
             combined_args["origin_str"] = full_name
 
+            # Remove any progress_reporter from the args as that will not transfer over the wire
+            if "progress_reporter" in combined_args:
+                del combined_args["progress_reporter"]
```

Comment:
Also need to remove the progress_reporter as an arg when reporting from here too.